fs2-google-pubsub-http/src/main/scala/com/permutive/pubsub/producer/http/internal/DefaultHttpPublisher.scala

```diff
@@ -165,8 +181,21 @@ private[http] object DefaultHttpPublisher {
   implicit final val MessageCodec: JsonValueCodec[Message] =
     JsonCodecMaker.make[Message](CodecMakerConfig)
 
-  implicit final def messageBundleCodec[G[_]: Foldable]: JsonValueCodec[MessageBundle[G]] =
-    JsonCodecMaker.make[MessageBundle[G]](CodecMakerConfig)
+  implicit final def messageBundleCodec[G[_]](implicit
+    Codec: JsonValueCodec[G[Message]]
+  ): JsonValueCodec[MessageBundle[G]] =
+    new JsonValueCodec[MessageBundle[G]] {
+      override def decodeValue(in: JsonReader, default: MessageBundle[G]): MessageBundle[G] = ???
```

Comment:
**question:** should this be implemented?

<sub>This comment follows the [**conventionalcomments.org**](https://conventionalcomments.org/) standard</sub>

Reply:
Yeah, to confirm we discussed this offline and I think this is OK. As Tim says we need to "define" this to make a valid instance, but we never actually need the decoder. It has been like this we first wrote the lib 👍